### PR TITLE
Pyiron atomistic split

### PR DESCRIPTION
**Pull request by niklassiemer**
Initializing the transition to pyiron_atomistic by renaming the pyiron directory to pyiron_atomistic.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -305,7 +305,7 @@
                 os.path.join(curdir, 'source/notebooks'))
 
 if 'readthedocs.org' in curdir:  # write config for readthedocs.org
-    with open(os.path.join(os.path.expanduser('~/.pyiron')), 'w') as f:
+    with open(os.path.join(os.path.expanduser('~/.pyiron_atomistic')), 'w') as f:
```
Comment (jan-janssen) [inline docs/conf.py:308]: I guess the name of the config file remains `~/.pyiron`.

**Comment by jan-janssen** [comment]
The error message is not very intuitive: 
```
======================================================================
1891
ERROR: test_run (interactive.test_scipyminimizer.TestSxExtOptInteractive)
1892
----------------------------------------------------------------------
1893
Traceback (most recent call last):
1894
  File "/home/runner/work/pyiron/pyiron/tests/interactive/test_scipyminimizer.py", line 32, in test_run
1895
    minim.run()
1896
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/generic/util.py", line 213, in decorated
1897
    return function(*args, **kwargs)
1898
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/job/generic.py", line 669, in run
1899
    self._run_if_new(debug=debug)
1900
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/job/generic.py", line 1336, in _run_if_new
1901
    self.validate_ready_to_run()
1902
  File "/home/runner/work/pyiron/pyiron/pyiron_atomistic/atomistics/job/interactivewrapper.py", line 89, in validate_ready_to_run
1903
    self.ref_job.validate_ready_to_run()
1904
AttributeError: 'NoneType' object has no attribute 'validate_ready_to_run'
1905

1906
======================================================================
1907
ERROR: test_run_pressure (interactive.test_scipyminimizer.TestSxExtOptInteractive)
1908
----------------------------------------------------------------------
1909
Traceback (most recent call last):
1910
  File "/home/runner/work/pyiron/pyiron/tests/interactive/test_scipyminimizer.py", line 47, in test_run_pressure
1911
    minim.run()
1912
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/generic/util.py", line 213, in decorated
1913
    return function(*args, **kwargs)
1914
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/job/generic.py", line 669, in run
1915
    self._run_if_new(debug=debug)
1916
  File "/usr/share/miniconda/envs/test/lib/python3.8/site-packages/pyiron_base/job/generic.py", line 1336, in _run_if_new
1917
    self.validate_ready_to_run()
1918
  File "/home/runner/work/pyiron/pyiron/pyiron_atomistic/atomistics/job/interactivewrapper.py", line 89, in validate_ready_to_run
1919
    self.ref_job.validate_ready_to_run()
1920
AttributeError: 'NoneType' object has no attribute 'validate_ready_to_run'
1921

1922
----------------------------------------------------------------------
```

**Comment by niklassiemer** [comment]
Actually, I think we have a problem due to  
https://github.com/pyiron/pyiron_base/blob/9fce7ef80c9e58a2a4e109ced403c3c371d7c292/pyiron_base/job/generic.py#L1050
where we reference pyiron_atomistic in base! I will raise an issue in pyiron_base.

File: docs/conf.py (hunk 1 above)
Comment (niklassiemer) [reply docs/conf.py:308]: Yes, it absolutely should!

**Comment by jan-janssen** [comment]
As the unit tests seem to be working now we should discuss the next steps:

- We should add a backwards compatibility interface at least for the project class, the atoms class and the functions commonly used to convert ASE structures and pymatgen structures to pyiron structures and back. 
- Once the backwards compatibility interface is added we should merge these changes.
- After the merge we can create the separate pyiron_atomistic repository.

**Comment by jan-janssen** [comment]
The test installation notebook fails:
```
Traceback (most recent call last):
355
  File "/usr/share/miniconda/envs/test/bin/papermill", line 10, in <module>
356
    sys.exit(papermill())
357
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 829, in __call__
358
    return self.main(*args, **kwargs)
359
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 782, in main
360
    rv = self.invoke(ctx)
361
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
362
    return ctx.invoke(self.callback, **ctx.params)
363
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 610, in invoke
364
    return callback(*args, **kwargs)
365
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
366
    return f(get_current_context(), *args, **kwargs)
367
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/cli.py", line 240, in papermill
368
    execute_notebook(
369
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 118, in execute_notebook
370
    raise_for_execution_errors(nb, output_path)
371
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 230, in raise_for_execution_errors
372
    raise error
373
papermill.exceptions.PapermillExecutionError: 
374
---------------------------------------------------------------------------
375
Exception encountered at "In [3]":
376
---------------------------------------------------------------------------
377
ModuleNotFoundError                       Traceback (most recent call last)
378
<ipython-input-3-cf39f0701255> in <module>
379
----> 1 import pyiron
380
      2 pyiron.__file__  # location where pyiron is installed
381

382
ModuleNotFoundError: No module named 'pyiron'
```

**Comment by jan-janssen** [comment]
As well as the structure notebook:
```
Traceback (most recent call last):
317
  File "/usr/share/miniconda/envs/test/bin/papermill", line 10, in <module>
318
    sys.exit(papermill())
319
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 829, in __call__
320
    return self.main(*args, **kwargs)
321
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 782, in main
322
    rv = self.invoke(ctx)
323
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
324
    return ctx.invoke(self.callback, **ctx.params)
325
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 610, in invoke
326
    return callback(*args, **kwargs)
327
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
328
    return f(get_current_context(), *args, **kwargs)
329
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/cli.py", line 240, in papermill
330
    execute_notebook(
331
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 118, in execute_notebook
332
    raise_for_execution_errors(nb, output_path)
333
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 230, in raise_for_execution_errors
334
    raise error
335
papermill.exceptions.PapermillExecutionError: 
336
---------------------------------------------------------------------------
337
Exception encountered at "In [8]":
338
---------------------------------------------------------------------------
339
ModuleNotFoundError                       Traceback (most recent call last)
340
<ipython-input-8-2a50d2c1ecbf> in <module>
341
----> 1 from pyiron import create_ase_bulk, create_surface
342

343
ModuleNotFoundError: No module named 'pyiron'
```

Annd the gamma surface notebook: 
```
Traceback (most recent call last):
185
  File "/usr/share/miniconda/envs/test/bin/papermill", line 10, in <module>
186
    sys.exit(papermill())
187
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 829, in __call__
188
    return self.main(*args, **kwargs)
189
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 782, in main
190
    rv = self.invoke(ctx)
191
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
192
    return ctx.invoke(self.callback, **ctx.params)
193
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/core.py", line 610, in invoke
194
    return callback(*args, **kwargs)
195
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/click/decorators.py", line 21, in new_func
196
    return f(get_current_context(), *args, **kwargs)
197
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/cli.py", line 240, in papermill
198
    execute_notebook(
199
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 118, in execute_notebook
200
    raise_for_execution_errors(nb, output_path)
201
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/papermill/execute.py", line 230, in raise_for_execution_errors
202
    raise error
203
papermill.exceptions.PapermillExecutionError: 
204
---------------------------------------------------------------------------
205
Exception encountered at "In [3]":
206
---------------------------------------------------------------------------
207
ModuleNotFoundError                       Traceback (most recent call last)
208
<ipython-input-3-9a3ec904e189> in <module>
209
----> 1 from pyiron.atomistics.master.parallel import AtomisticParallelMaster
210
      2 from pyiron_base.master.parallel import JobGenerator
211

212
ModuleNotFoundError: No module named 'pyiron'
```

**Comment by jan-janssen** [comment]
I try to fix the notebooks in https://github.com/pyiron/pyiron_atomistic/pull/2

**Comment by jan-janssen** [comment]
The conda package is now available https://anaconda.org/conda-forge/pyiron_atomistics/

**Pull request closed without merging**